app/main.py

```diff
@@ -1,25 +1,71 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.max_amount = max_amount
+        self.min_amount = min_amount
+
+    def __set_name__(
+        self,
+        owner: type[object],
+        name: str
+    ) -> None:
+        self.public_name = name
+        self.private_name = "_" + name
+
+    def __get__(
+        self,
+        instance: object,
+        owner: type[object]
+    ) -> int:
+        return getattr(instance, self.private_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        assert self.min_amount <= value <= self.max_amount
+        setattr(instance, self.private_name, value)
```

Comment:
I think it's better to raise here `ValueError`. Also you can add validation for type of the value and raise TypeError